### PR TITLE
Package resource type

### PR DESCRIPTION
**Pull request by pulquero**
Refactored to create a 'package' resource type so it is possible to install multiple jdk versions.

Example:
	jdk_oracle::package {
	'jdk6':
		version => '6',
		create_symlink => false,
		default_java => false;
	'jdk7':
		version => '7'
	}

Changes:
contents of init.pp moved to package.pp and resource titles made unique by addition of _${version} where necessary.

### CODE REVIEW

**Comment by tylerwalts** [comment]
Thanks, generally this is a great idea, especially for use cases like configuring a CI build server.  There is another fork out there that just converted it from a class to a define in the main init.pp, but no PR was done:

https://github.com/burdara/puppet-jdk_oracle/blob/master/manifests/init.pp

A few feedback items:
* For the naming, I feel like `package.pp` is somewhat misleading since we are installing intentionally from tarball vs using a OS package.
    * Could we call it something more generic, like `install.pp`?
* Tests are not passing
    * Not sure right now if it was caused by your commits or some other issue?

File: manifests/init.pp [hunk 1]
```diff
@@ -55,283 +55,17 @@
...
-        source  => "${install_dir}/${jce_dir}/US_export_policy.jar",
-        mode    => '0644',
-        owner   => 'root',
-        group   => 'root',
-        require => Exec['extract_jce'],
-      }
-
-    }
-
+  jdk_oracle::package { 'jdk_oracle':
```
Comment (tylerwalts) [inline manifests/init.pp]: Should this now be:  `jdk_oracle::install` ?

**Comment by tylerwalts** [comment]
This seems like a breaking change and people may have to update their references to use it as a define now vs a class (exa:  hiera users), so it feels like this should be a major 2.0.0 release, to make it clear that change will happen.  Agree/disagree?

File: manifests/init.pp (hunk 1 above)
Comment (pulquero) [inline manifests/init.pp]: Yes.

**Comment by pulquero** [comment]
I disagree, the original class definition is still there, so anybody still using it should not be affected. But, I'm not a puppet expert, so don't know if I'm overlooking something.

**Comment by tylerwalts** [comment]
Good point, we have the original class with parameters which remain unchanged, so it would be the implementer's option to call the define directly, or not.

**Pull request merged**